scipy/special/wright.hh

```diff
@@ -1,6 +1,7 @@
 #ifndef WRIGHT_HH
 #define WRIGHT_HH
 
+#include <pyconfig.h>
```

Comment:
Instead of this, can you move `#include <Python.h>` from line 75 to up here?

Reply:
Ah sorry, my comment was confusing. What I meant was to make a change in `wright.cc`, moving `#include <Python.h>` above `#include "wright.hh`. And then this change is unnecessary.

Reply:
What's the reasoning for preferring to keep `pyconfig.h` out of the header?  Ease of comparison with external source?

Also changed `_wright.h` the same way.

Reply:
It's that `pyconfig.h` isn't actually used in these files. It's a side effect of its contents (`_XOPEN_SOURCE` in this case) that you're after. Which is a sign that something else is wrong. Which is pretty clear cut here. `Python.h` must always be imported before any standard library C/C++ headers, not doing so is a bug. And once the bug is fixed, there's no need for `pyconfig.h`.

Reply:
Unfortunately there's no static checker to enforce the header ordering - if there was one, I'd gladly use it. This bites us too often.